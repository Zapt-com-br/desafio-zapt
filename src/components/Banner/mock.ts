export default {
  title: 'Bem vinda Polyane,',
  contactNumber: '11910898780'
}
