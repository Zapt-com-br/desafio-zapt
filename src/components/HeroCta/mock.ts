const heroTitle = 'Bem vindo Marcio,'
const heroSubtitle =
  'Para você aproveitar nossas oportunidades primeiro você precisa estar dentro de um grupo, estamos aqui para lhe ajudar.'
const ctaTitle = 'O que você prefere?'
const firstOptionLabel = 'Entrar em um grupo'
const secondOptionLabel = 'Criar um grupo'

export default {
  heroTitle,
  ctaTitle,
  heroSubtitle,
  firstOptionLabel,
  secondOptionLabel
}
