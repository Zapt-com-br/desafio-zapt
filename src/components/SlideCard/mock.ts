const userComment = {
  content:
    '”Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal”',
  user: {
    name: 'Patricia Godoy',
    location: 'São Paulo • São Paulo',
    icon: '/img/users-comments/woman-icon.png'
  },
  backgroundImage: '/img/users-comments/woman-image.png'
}

export default userComment
