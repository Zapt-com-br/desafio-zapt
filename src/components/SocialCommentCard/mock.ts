export default {
  content:
    '”Não da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um só dia... O pessoal aqui de casa ficou viciado de um nível que terei que comprar toda semana... Assim não tem academia que secura rsrsrsrs Obrigado Zapt.”',
  user: {
    profileIcon: '/img/users-icon/user-icon1.png',
    username: '@ferri-gui'
  }
}
