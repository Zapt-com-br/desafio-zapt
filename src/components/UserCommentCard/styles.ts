import NextImage from 'next/image';
import styled, { css } from 'styled-components';
import media from 'styled-media-query';

export const Wrapper = styled.div`
  max-width: 58.8rem;
  max-height: 40rem;
  position: relative;
`;

export const ImageWrapper = styled.div`
  position: relative;
  width: 100%;
  height: 30rem;
  filter: brightness(70%);

  ${media.greaterThan('medium')`
    height: 40rem;
  `}
`;

export const Image = styled(NextImage)`
  ${({ theme }) => css`
    border-radius: ${theme.border.radius};
  `}
`;

export const Content = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  padding: 2.4rem;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
`;

export const Comment = styled.p`
  ${({ theme }) => css`
    width: 85%;
    margin-bottom: 1.6rem;
    font-size: ${theme.font.sizes.xsmall};

    ${media.greaterThan('small')`
      font-size: ${theme.font.sizes.medium};
    `}

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}

    ${media.greaterThan('huge')`
      font-size: ${theme.font.sizes.xlarge};
    `}
  `}
`;

export const Quote = styled.q`
  ${({ theme }) => css`
    color: ${theme.colors.white};
  `}
`;
