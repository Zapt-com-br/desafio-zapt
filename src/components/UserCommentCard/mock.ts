import { userInfo } from 'components/UserInfo/mock';

import { UserCommentCardProps } from '.';

export const userComment: UserCommentCardProps = {
  imageBackgroundUrl: 'https://i.imgur.com/ci2Eulu.png',
  imageBackgroundAlt: 'Senhora cortante legumes',
  comment:
    'Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal',
  user: userInfo,
};

export const anotherUserComment: UserCommentCardProps = {
  imageBackgroundUrl: 'https://i.imgur.com/brAYg9C.png',
  imageBackgroundAlt: 'Homem e mulher lavando alimentos',
  comment:
    'Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...',
  user: {
    avatarUrl: 'https://i.imgur.com/jn8VYLl.png',
    name: 'Marcinho Ribeiro',
    city: 'Belo Horizonte',
    state: 'Minas Gerais',
  },
};
