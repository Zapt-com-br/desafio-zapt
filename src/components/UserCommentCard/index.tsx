import UserInfo, { UserInfoProps } from 'components/UserInfo';

export type UserCommentCardProps = {
  imageBackgroundUrl: string;
  imageBackgroundAlt: string;
  comment: string;
  user: UserInfoProps;
};

import * as S from './styles';

const UserCommentCard = ({
  imageBackgroundUrl,
  imageBackgroundAlt,
  comment,
  user,
}: UserCommentCardProps) => (
  <S.Wrapper>
    <S.ImageWrapper>
      <S.Image
        src={imageBackgroundUrl}
        alt={imageBackgroundAlt}
        layout="fill"
        placeholder="blur"
        objectFit="cover"
        blurDataURL={imageBackgroundUrl}
      />
    </S.ImageWrapper>

    <S.Content>
      <S.Comment>
        <S.Quote>{comment}</S.Quote>
      </S.Comment>

      <UserInfo
        avatarUrl={user.avatarUrl}
        name={user.name}
        city={user.city}
        state={user.state}
      />
    </S.Content>
  </S.Wrapper>
);

export default UserCommentCard;
