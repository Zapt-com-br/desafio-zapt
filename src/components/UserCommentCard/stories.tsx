import { Story, Meta } from '@storybook/react/types-6-0';

import UserCommentCard, { UserCommentCardProps } from '.';
import { userComment } from './mock';

export default {
  title: 'UserCommentCard',
  component: UserCommentCard,
} as Meta;

export const Default: Story<UserCommentCardProps> = (args) => (
  <UserCommentCard {...args} />
);

Default.args = { ...userComment };
