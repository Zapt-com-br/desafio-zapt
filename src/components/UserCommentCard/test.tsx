import { render, screen } from 'utils/test-utils';

import UserCommentCard from '.';

import { userComment } from './mock';

describe('<UserCommentCard />', () => {
  it('should math snapshot', () => {
    const { container } = render(<UserCommentCard {...userComment} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with background image', () => {
    render(<UserCommentCard {...userComment} />);

    expect(
      screen.getByRole('img', { name: userComment.imageBackgroundAlt })
    ).toHaveAttribute('src', userComment.imageBackgroundUrl);
  });

  it('should render with comment', () => {
    render(<UserCommentCard {...userComment} />);

    expect(screen.getByText(userComment.comment)).toBeInTheDocument();
  });
});
