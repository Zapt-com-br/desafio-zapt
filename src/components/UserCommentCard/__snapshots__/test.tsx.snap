// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCommentCard /> should math snapshot 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  width: 4rem;
  height: 4rem;
  border-radius: 50%;
  border: 0.15rem solid #ffffff;
}

.c7 {
  color: #ffffff;
  margin-left: 1.6rem;
}

.c8 {
  font-size: 1.6rem;
  font-weight: 400;
}

.c9 {
  font-size: 1.2rem;
  font-weight: 400;
}

.c0 {
  max-width: 58.8rem;
  max-height: 40rem;
  position: relative;
}

.c1 {
  border-radius: 0.4rem;
  width: 100%;
  height: 100%;
  -webkit-filter: brightness(70%);
  filter: brightness(70%);
}

.c2 {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  padding: 2.4rem;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 {
  width: 85%;
  margin-bottom: 1.6rem;
  font-size: 1.2rem;
}

.c4 {
  color: #ffffff;
}

@media (min-width:450px) {
  .c3 {
    font-size: 1.6rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 1.8rem;
  }
}

@media (min-width:1440px) {
  .c3 {
    font-size: 2.0rem;
  }
}

<div
  class="c0"
>
  <img
    alt="Senhora cortante legumes"
    class="c1"
    src="https://i.imgur.com/ci2Eulu.png"
  />
  <div
    class="c2"
  >
    <p
      class="c3"
    >
      <q
        class="c4"
      >
        Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal
      </q>
    </p>
    <div
      class="c5"
    >
      <img
        alt="Patricia Godoy"
        class="c6"
        src="https://i.imgur.com/8RP8XA9.png"
      />
      <div
        class="c7"
      >
        <h3
          class="c8"
        >
          Patricia Godoy
        </h3>
        <p
          class="c9"
        >
          São Paulo
           • 
          São Paulo
        </p>
      </div>
    </div>
  </div>
</div>
`;
